src/main/java/com/aws/greengrass/deployment/bootstrap/BootstrapManager.java

```diff
@@ -478,8 +482,10 @@ public void deleteBootstrapTaskList(Path persistedTaskFilePath) throws IOExcepti
             logger.atError().log("No bootstrap task list to delete: the provided file path was null");
             return;
         }
-        logger.atInfo().kv("filePath", persistedTaskFilePath).log("Deleting bootstrap task list");
-        Files.deleteIfExists(persistedTaskFilePath);
+        if (Files.exists(persistedTaskFilePath)) {
+            logger.atInfo().kv("filePath", persistedTaskFilePath).log("Deleting bootstrap task list");
+            Files.delete(persistedTaskFilePath);
```

Comment:
Use delete if exists and use the return value to prevent time of check to time of use race.

Reply:
updated